assembly/src/lib.rs

```diff
@@ -68,6 +68,20 @@ impl ops::Deref for ProcedureId {
     }
 }
 
+impl fmt::Display for ProcedureId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut res = String::from("[");
+        for i in 0..self.0.len() - 1 {
+            res.push_str(&self.0[i].to_string());
+            res.push(',');
+            res.push(' ');
+        }
+        res.push_str(&self.0[self.0.len() - 1].to_string());
+        res.push(']');
+        write!(f, "{res}")
```

Comment:
I'd prefer to print it out as a hexadecimal string. You can probably use the code similar to the one [here](https://github.com/maticnetwork/miden/blob/next/core/src/program/blocks/call_block.rs#L67).